### PR TITLE
add tip basics declarations

### PR DESCRIPTION
**Pull request by adaniloff**
Hello. 
Sorry for english mistakes if any, it's not my native language.

Feel free to comment and tell me what's wrong, if anything :).

First pull request on github by the way, please be gentle with me :D.

### CODE REVIEW

File: _posts/en/2016-02-15-basics-declarations.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+;(() => { 
+  var x = y = 2 // == var x; y = 2;
+  console.log('2.0:', `x = ${x}, y = ${y}`)
+})()
+console.log('--> 2.1:', `x = ${x}, y = ${y}`)
+
+// Will print
+//2.0: x = 2, y = 2
+//--> 2.1: x = 1, y = 2
```
Comment (kurtextrem) [inline _posts/en/2016-02-15-basics-declarations.md]: No idea what 2.1 is.

File: _posts/en/2016-02-15-basics-declarations.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+```js
+var y, x = y = 1 //== var x; var y; x = y = 1
+console.log('--> 1:', `x = ${x}, y = ${y}`)
+
+// Will print
+//--> 1: x = 1, y = 1
+
+;(() => { 
+  var x = y = 2 // == var x; y = 2;
```
Comment (kurtextrem) [inline _posts/en/2016-02-15-basics-declarations.md]: the == is wrong.

File: _posts/en/2016-02-15-basics-declarations.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  console.log('3.0:', `x = ${x}, y = ${y}`)
+})()
+console.log('--> 3.1:', `x = ${x}, y = ${y}`)
+
+// Will print
+//3.0: x = undefined, y = 3
+//--> 3.1: x = 1, y = 2
+
+;(() => { 
+  var y, x = y = 4 // == var x; var y; x = y = 3
```
Comment (kurtextrem) [inline _posts/en/2016-02-15-basics-declarations.md]: The == is wrong.

File: _posts/en/2016-02-15-basics-declarations.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  console.log('4.0:', `x = ${x}, y = ${y}`)
+})()
+console.log('--> 4.1:', `x = ${x}, y = ${y}`)
+
+// Will print
+//4.0: x = 4, y = 4
+//--> 4.1: x = 1, y = 2
+
+x = 5 // == x = 5
+console.log('--> 5:', `x = ${x}, y = ${y}`)
```
Comment (kurtextrem) [inline _posts/en/2016-02-15-basics-declarations.md]: Where does the y come from?

**Comment by kurtextrem** [comment]
Please take a look at it again. I've commented inline.

Also I'm not sure what tip you're trying to get over. Can you please explain?

Thank you,
Jacob

**Comment by adaniloff** [comment]
First, Thank you for replying to me.
I'm trying to explain which differences there are with theses different variable declarations.

EDIT : the idea is to see how x and y change with those different declarations

Could you explain to me what's wrong please? (I don't see any problem... ?) 

Again, excuse me for my poor english.

You can test the script and see console.log results by clicking on the following link : [here](https://babeljs.io/repl/#?experimental=false&evaluate=true&loose=false&spec=false&code=var%20y%2C%20x%20%3D%20y%20%3D%201%20%2F%2F%3D%3D%20var%20x%3B%20var%20y%3B%20x%20%3D%20y%20%3D%201%0Aconsole.log%28'--%3E%201%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%0A%2F%2F%20Will%20print%0A%2F%2F--%3E%201%3A%20x%20%3D%201%2C%20y%20%3D%201%0A%0A%3B%28%28%29%20%3D%3E%20%7B%20%0A%20%20var%20x%20%3D%20y%20%3D%202%20%2F%2F%20%3D%3D%20var%20x%3B%20y%20%3D%202%3B%0A%20%20console.log%28'2.0%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%7D%29%28%29%0Aconsole.log%28'--%3E%202.1%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%0A%2F%2F%20Will%20print%0A%2F%2F2.0%3A%20x%20%3D%202%2C%20y%20%3D%202%0A%2F%2F--%3E%202.1%3A%20x%20%3D%201%2C%20y%20%3D%202%0A%0A%3B%28%28%29%20%3D%3E%20%7B%20%0A%20%20var%20x%2C%20y%20%3D%203%20%2F%2F%20%3D%3D%20var%20x%3B%20var%20y%20%3D%203%3B%0A%20%20console.log%28'3.0%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%7D%29%28%29%0Aconsole.log%28'--%3E%203.1%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%0A%2F%2F%20Will%20print%0A%2F%2F3.0%3A%20x%20%3D%20undefined%2C%20y%20%3D%203%0A%2F%2F--%3E%203.1%3A%20x%20%3D%201%2C%20y%20%3D%202%0A%0A%3B%28%28%29%20%3D%3E%20%7B%20%0A%20%20var%20y%2C%20x%20%3D%20y%20%3D%204%20%2F%2F%20%3D%3D%20var%20x%3B%20var%20y%3B%20x%20%3D%20y%20%3D%203%0A%20%20console.log%28'4.0%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%7D%29%28%29%0Aconsole.log%28'--%3E%204.1%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%0A%2F%2F%20Will%20print%0A%2F%2F4.0%3A%20x%20%3D%204%2C%20y%20%3D%204%0A%2F%2F--%3E%204.1%3A%20x%20%3D%201%2C%20y%20%3D%202%0A%0Ax%20%3D%205%20%2F%2F%20%3D%3D%20x%20%3D%205%0Aconsole.log%28'--%3E%205%3A'%2C%20%60x%20%3D%20%24%7Bx%7D%2C%20y%20%3D%20%24%7By%7D%60%29%0A%0A%2F%2F%20Will%20print%0A%2F%2F--%3E%205%3A%20x%20%3D%205%2C%20y%20%3D%202)

**Comment by kurtextrem** [comment]
Ah, I see. It is one context and they live in the same environment.
Now I see what you're trying to get over.

Could you please add more comments to the tip? E.g. after the 2nd output:
"As you can see, the code has only changed the global `y`, as we haven't declared the variable in the closure."
After the 3rd:
"Now we declare both variables through `var`. Meaning they only live in the context of the closure."
After the 4th:
"Both variables have been declared using `var` and only after that we've set their values. As local > global, `x` and `y` are local in the closure, meaning the global `x` and `y` are untouched."
That makes it more obvious to what exactly is going on.

Also adding the Babel link is so helpful. Can you please also add it?

The comment in line 28 is still wrong.

**Comment by adaniloff** [comment]
Ahh, thanks for the L.28; I hadn't noticed until I read your last comment!
Also, good idea too add explanations, I thought what I was trying to do was obvious but I see now it is not, thanks to you.

I'm doing all that, thanks man!

**Comment by adaniloff** [comment]
Should / could I mention you ? For your help :) ? 
Thanks again !

File: _posts/en/2016-02-15-basics-declarations.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+tip-number: 46
+tip-username: adaniloff 
+tip-username-profile: https://github.com/adaniloff
+tip-tldr: Understand and work with declarations.
+
+categories:
+    - en
+---
+
+Below, different ways to declare variables in javascript.
```
Comment (kurtextrem) [inline _posts/en/2016-02-15-basics-declarations.md]: Should be "JavaScript".

**Comment by kurtextrem** [comment]
You decide, It's all your tip. I'm just trying to help :) 

I've added one last comment - and I think it would be better to add ` to the x, y and var in the explanations (to understand that we're talking about code)

**Comment by adaniloff** [comment]
Damn. I changed my setup env and can't figure how to make thoses ` work now. I'm on it!
EDIT : ok, that's good now !

**Comment by kurtextrem** [comment]
Seems good to me! :+1:

**Comment by adaniloff** [comment]
Yeah! Thanks for your help :)

Btw the last commit I did remove the special thanks I'd written.
Gonna do it again.

Should I do something else for the pull request? Like redo a pull request or so ...? First time doing it so I don't really know :).

**Comment by kurtextrem** [comment]
I guess you have to change the file name, as today is the 02-16.

Apart from that, @loverajoel ping

**Comment by adaniloff** [comment]
Damn, you're right, thanks again (what would I do without you ahah).
Doing this @ + nickname ping, will trigger an alert for him ? Just to know :)

**Comment by kurtextrem** [comment]
He will get a notification that he was mentioned :)

**Comment by adaniloff** [comment]
Got it :)

**Comment by loverajoel** [comment]
good tip! thanks for your tip @adaniloff and for your review @kurtextrem !

**Pull request merged**